app/gui2/src/components/GraphEditor.vue

```diff
@@ -51,24 +52,53 @@ const componentBrowserUsage = ref<Usage>({ type: 'newNode' })
 const suggestionDb = useSuggestionDbStore()
 const interaction = provideInteractionHandler()
 
+/// === UI Messages and Errors ===
 function initStartupToast() {
   const startupToast = toast.info('Initializing the project. This can take up to one minute.', {
     autoClose: false,
   })
   projectStore.firstExecution.then(() => {
-    if (startupToast != null) {
-      toast.remove(startupToast)
-    }
+    toast.dismiss(startupToast)
   })
   onUnmounted(() => {
-    if (startupToast != null) {
-      toast.remove(startupToast)
-    }
+    toast.dismiss(startupToast)
+  })
+}
+
+function initiConnectionLostToast() {
+  let connectionLostToast = 'connectionLostToast'
+  document.addEventListener(
+    ProjectManagerEvents.loadingFailed,
+    () => {
+      toast.error('Lost connection to Language Server.', {
+        autoClose: false,
+        toastId: connectionLostToast,
+      })
+    },
+    { once: true },
+  )
+  onUnmounted(() => {
+    toast.dismiss(connectionLostToast)
   })
 }
 
+projectStore.lsRpcConnection.catch((err) => {
```

Comment:
you can provide the `catch` callback as the second argument to the `.then()` below.